be/src/agent/task_worker_pool.cpp

```diff
@@ -1693,4 +1697,43 @@ void TaskWorkerPool::_storage_update_storage_policy_worker_thread_callback() {
     }
 }
 
+void TaskWorkerPool::_push_cooldown_conf_worker_thread_callback() {
+    while (_is_work) {
+        TAgentTaskRequest agent_task_req;
+        TPushCooldownConfReq push_cooldown_conf_req;
+        {
+            std::unique_lock<std::mutex> worker_thread_lock(_worker_thread_lock);
+            while (_is_work && _tasks.empty()) {
+                _worker_thread_condition_variable.wait(worker_thread_lock);
+            }
+            if (!_is_work) {
+                return;
+            }
+
+            agent_task_req = _tasks.front();
+            push_cooldown_conf_req = agent_task_req.push_cooldown_conf;
+            _tasks.pop_front();
+        }
+        int64_t tablet_id = push_cooldown_conf_req.tablet_id;
+        TabletSharedPtr tablet = StorageEngine::instance()->tablet_manager()->get_tablet(tablet_id);
+        if (tablet.get() == nullptr) {
+            std::stringstream ss;
+            ss << "failed to get tablet. tablet_id=" << tablet_id;
+            LOG(WARNING) << ss.str();
+            return;
+        }
+        if (push_cooldown_conf_req.cooldown_term > tablet->tablet_meta()->cooldown_term()) {
```

Comment:
update `cooldown_term` and `cooldown_replica_id` MUST be an atomic operation. I suggest to implement a `void Tablet::update_cooldown_conf(int64_t cooldown_term, int64_t cooldown_replica_id)` function.